sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -186,13 +187,18 @@ private[sql] class SessionState(sparkSession: SparkSession) {
   }
 
   /**
-   * Analyzes the given table in the current database to generate statistics, which will be
+   * Analyzes the given table in the current database to generate table-level statistics, which
+   * will be used in query optimizations.
+   */
+  def analyzeTable(tableIdent: TableIdentifier, noscan: Boolean = true): Unit = {
+    AnalyzeTableCommand(tableIdent, noscan).run(sparkSession)
+  }
+
+  /**
+   * Analyzes the given columns in the table to generate column-level statistics, which will be
    * used in query optimizations.
-   *
-   * Right now, it only supports catalog tables and it only updates the size of a catalog table
-   * in the external catalog.
    */
-  def analyze(tableName: String, noscan: Boolean = true): Unit = {
```

Comment:
Now AnalyzeTableCommand receives TableIdentifier instead of String tableName as parameter, as @hvanhovell suggested in [comment](https://github.com/apache/spark/pull/15090#r78717769)